assets/terraform-modules/aws/flatcar-linux/kubernetes/cl/controller.yaml.tmpl

```diff
@@ -55,32 +52,30 @@ systemd:
         [Service]
         ConditionPathExists=/etc/kubernetes/kubeconfig
         EnvironmentFile=/etc/kubernetes/kubelet.env
-        Environment="RKT_RUN_ARGS=--uuid-file-save=/var/cache/kubelet-pod.uuid \
-          --volume=resolv,kind=host,source=/etc/resolv.conf \
-          --mount volume=resolv,target=/etc/resolv.conf \
-          --volume var-lib-cni,kind=host,source=/var/lib/cni \
-          --mount volume=var-lib-cni,target=/var/lib/cni \
-          --volume var-lib-calico,kind=host,source=/var/lib/calico \
-          --mount volume=var-lib-calico,target=/var/lib/calico \
-          --volume opt-cni-bin,kind=host,source=/opt/cni/bin \
-          --mount volume=opt-cni-bin,target=/opt/cni/bin \
-          --volume var-log,kind=host,source=/var/log \
-          --mount volume=var-log,target=/var/log \
-          --volume etc-cni-netd,kind=host,source=/etc/cni/net.d \
-          --mount volume=etc-cni-netd,target=/etc/cni/net.d \
-          --insecure-options=image"
-        ExecStartPre=/bin/mkdir -p /opt/cni/bin
-        ExecStartPre=/bin/mkdir -p /etc/kubernetes/manifests
-        ExecStartPre=/bin/mkdir -p /etc/cni/net.d
-        ExecStartPre=/bin/mkdir -p /etc/kubernetes/checkpoint-secrets
-        ExecStartPre=/bin/mkdir -p /etc/kubernetes/inactive-manifests
-        ExecStartPre=/bin/mkdir -p /var/lib/cni
-        ExecStartPre=/bin/mkdir -p /var/lib/calico
         ExecStartPre=/bin/mkdir -p /var/lib/kubelet/volumeplugins
         ExecStartPre=/usr/bin/bash -c "grep 'certificate-authority-data' /etc/kubernetes/kubeconfig | awk '{print $2}' | base64 -d > /etc/kubernetes/ca.crt"
         ExecStartPre=/etc/kubernetes/configure-kubelet-cgroup-driver
-        ExecStartPre=-/usr/bin/rkt rm --uuid-file=/var/cache/kubelet-pod.uuid
-        ExecStart=/usr/lib/coreos/kubelet-wrapper \
+        ExecStart=docker run \
+          --network=host \
+          --pid=host \
+          --privileged \
+          -v /dev:/dev:rw \
+          -v /etc/cni/net.d:/etc/cni/net.d:rw \
+          -v /etc/kubernetes:/etc/kubernetes:ro \
+          -v /etc/machine-id:/etc/machine-id \
+          -v /lib/modules:/lib/modules \
+          -v /run:/run:rw \
```

Comment:
Why rw just to a specific path only for the kubelet files?

Reply:
It uses unspecified files from here and a lock file:

```console
# lsof | grep '/run' | grep kubelet                                                                                                           
kubelet   10931                   root    5uW     REG               0,21         0      30352 /run/lock/kubelet.lock        
kubelet   10931                   root   10u     unix 0xffff9fd3b65b6400       0t0     311644 /var/run/661904757 type=STREAM
kubelet   10931                   root   14u     unix 0xffff9fd3b65b4c00       0t0     311194 /var/run/661904757 type=STREAM
kubelet   10931                   root   20u     unix 0xffff9fd3b65b5400       0t0     311195 /var/run/661904757 type=STREAM
kubelet   10931                   root   31u     unix 0xffff9fd2b4d93800       0t0     311748 /var/run/661904757 type=STREAM
kubelet   10931 10948             root    5uW     REG               0,21         0      30352 /run/lock/kubelet.lock        
kubelet   10931 10948             root   10u     unix 0xffff9fd3b65b6400       0t0     311644 /var/run/661904757 type=STREAM
kubelet   10931 10948             root   14u     unix 0xffff9fd3b65b4c00       0t0     311194 /var/run/661904757 type=STREAM
kubelet   10931 10948             root   20u     unix 0xffff9fd3b65b5400       0t0     311195 /var/run/661904757 type=STREAM
kubelet   10931 10948             root   31u     unix 0xffff9fd2b4d93800       0t0     311748 /var/run/661904757 type=STREAM
...
```

Reply:
Oh, and this path can't be set? If we can lock it down more, IMHO seems better. But we were running like this before with rkt, so no biggie :)

Reply:
Also, `/run/lock` must be shared via host path between host kubelet and self-hosted one. But `/var/run` is separate and only in the container, right?

Reply:
Between `/var` and `/var/run`, one is symlink of other